packages/oidc-client/src/lib/index.ts

```diff
@@ -0,0 +1,136 @@
+import { iFrameManager } from '@forgerock/iframe-manager';
+import {
+  createAuthorizeUrl,
+  GetAuthorizationUrlOptions,
+  getStoredAuthUrlValues,
+} from '@forgerock/sdk-oidc';
+import { store } from './store.js';
+import { fetchWellKnownConfig } from './wellknown.slice.js';
+
+interface OIDCConfig {
+  prefix?: string;
+  skipBackgroundRequest: boolean;
+  serverConfig: {
+    wellknown: string;
+  };
+  responseType: 'code' | 'token';
+}
+
+interface AuthorizeSuccessResponse {
+  code: string;
+  state: string;
+}
+
+interface AuthorizeUrlResponse {
+  authorizeUrl: string;
+}
+
+interface AuthorizeErrorResponse {
+  error: string;
+  error_description?: string;
+  state?: string;
+}
+
+type AuthorizeResponse = AuthorizeSuccessResponse | AuthorizeUrlResponse | AuthorizeErrorResponse;
+
+export type { AuthorizeSuccessResponse, AuthorizeErrorResponse, AuthorizeResponse, OIDCConfig };
+
+export function initialize(config: OIDCConfig) {
+  const iframeMgr = iFrameManager();
+
+  if (!config?.serverConfig?.wellknown) {
+    return {
+      error: 'requires a wellknown url initializing this factory.',
+    };
+  }
+
+  const wellKnownUrl = config.serverConfig.wellknown;
+
+  return {
+    authorize: async (
+      options: GetAuthorizationUrlOptions,
+      timeout = 3000,
+    ): Promise<AuthorizeResponse | { error: string }> => {
+      const { data, error } = await store.dispatch(
+        fetchWellKnownConfig.endpoints.fetchWellKnownConfig.initiate(wellKnownUrl),
+      );
+
+      if (error || !data) {
+        return {
+          error: `Error fetching wellknown config`,
+        };
+      }
+
+      const authorizePath = data.authorization_endpoint;
+      const authorizeUrl = await createAuthorizeUrl(authorizePath, options);
+
+      if (config.skipBackgroundRequest) {
```

Comment:
if we want to skip a background request, I wasn't reallly sure if we just want to let the user make the request or if we want to do. a `get` ourselves

Reply:
https://github.com/ForgeRock/ping-javascript-sdk/pull/342/commits/47dcce93df15af32562f094a412eea0ef23e8c66#diff-0a9757ba8423bb24190c39c3a5926baaf2697edeaa229ea49110c86dbd5304d8R57 in this commit you can see I built out the reducer and slice and do the request just wasn't really sure what way to approach it